File: Liskovs/compliant/UserService.java
public class UserService {
    public void setAge(int age) {
        if (age < 0) {
            throw new IllegalArgumentException("Age must be non-negative");
        }
        System.out.println("Age set to " + age);
    }
    public static void main(String[] args) {
        UserService service = new AdultUserService();
        service.setAge(10);
    }

}

class AdultUserService extends UserService {
    @Override
    public void setAge(int age) {
        super.setAge(age);

        if (age >= 18) {
            System.out.println("User is an adult");
        } else {
            System.out.println("User is a minor");
        }
    }
}

File: Liskovs/compliant/UserService.cs
using System;

public class UserService
{
    public virtual void SetAge(int age)
    {
        if (age < 0)
        {
            throw new ArgumentException("Age must be non-negative");
        }

        Console.WriteLine("Age set to " + age);
    }

    public static void Main(string[] args)
    {
        UserService service = new AdultUserService();
        service.SetAge(10);
    }
}

public class AdultUserService : UserService
{
    public override void SetAge(int age)
    {
        base.SetAge(age);

        if (age >= 18)
        {
            Console.WriteLine("User is an adult");
        }
        else
        {
            Console.WriteLine("User is a minor");
        }
    }
}
